### ISSUE TITLE
Reported method-parameters in contextual help seems to appear in random order

### ISSUE THREAD

**Issue by josteink** [razzmatazz/csharp-language-server#6]
See example below:

![image](https://user-images.githubusercontent.com/411338/141085261-886e1571-b69f-42a6-836c-56626ebda8cd.png)

The cursor is on a call to the `Url.Action(string action, string controller, object values)` overload.

The *documentation* provided seems to be for this overload and is quite good.

The expanded *parameter-list* however is very confusing because:

* lots of parameters
* parameter names not shown in function signature above
* to make matters worse this is an *extension-method* so it has one "hidden" parameter.
* and all in all the parameters does not seem to appear in declared order (instance, action, parameter, values)

As a user I would expect the following order:

* Extension method target (if shown at all)
* First param (action)
* Second param (controller)
* Third param (values)

Are there any particular reason the parameters are currently presented as they are?

**Comment by razzmatazz** [comment]
Oh.. I didn't fix the parameter visibility problem and the issue with extension problems.

**Comment by josteink** [comment]
That’s totally ok. Thanks for responding so quickly!

Just having the parameters in proper order makes this function magnitudes more user friendly. This is a good improvement in its own!

**Comment by razzmatazz** [comment]
> That’s totally ok. Thanks for responding so quickly!
> 
> Just having the parameters in proper order makes this function magnitudes more user friendly. This is a good improvement in its own!

I use this in my day job so every nuisance hurts me too :)

**Comment by josteink** [comment]
Yeah. Me too. Trying this one out for now. 

It seems to start up faster than Omnisharp, so I’m willing to give it a chance to see how it works out 🙂

**Comment by razzmatazz** [comment]
> Yeah. Me too. Trying this one out for now.
> 
> It seems to start up faster than Omnisharp, so I’m willing to give it a chance to see how it works out slightly_smiling_face

I noticed that too.. it seems, from looking at omnisharp impl, is that for some reason (compat? -- omnisharp has a long history and much better support for CSX, Unity et.c.) they load a project one-by-one. Whereas csharp-ls uses .sln file (if it finds a single one at the root dir) -- to slurp it with roslyn at once. Roslyn for some reason likes  .sln a lot from perf perspective :)

And then we're running on a much better JIT in csharp-ls, whereas omnisharp (again for compat) uses mono which is not that performant. I saw they have a PR coming where omnisharp will convert itself to .NET runtime-proper from mono.

I was trying to work with omnisharp for a long time, but getting my PRs merged was pain, some of them are hanging there for half-a-year and +.. So writing my own LSP gets me faster dev cycle, even if I don't support all the things omnisharp does, just the things I personally need :)
